lib/resources/tasks/build.js

```diff
@@ -1,26 +1,40 @@
 import gulp from 'gulp';
+import { CLIOptions, build as buildCLI } from 'aurelia-cli';
 import transpile from './transpile';
 import processMarkup from './process-markup';
 import processCSS from './process-css';
 import copyFiles from './copy-files';
-import {build} from 'aurelia-cli';
+import watch from './watch';
 import project from '../aurelia.json';
 
-export default gulp.series(
+let build = gulp.series(
   readProjectConfiguration,
-  gulp.parallel(
-    transpile,
+  gulp.parallel(transpile,
     processMarkup,
     processCSS,
-    copyFiles
-  ),
+    copyFiles),
   writeBundles
 );
 
+let main;
+
+// Only watch if we're the actual task run, not if we're 
+// called from run or other task
+if (CLIOptions.hasFlag("watch") && CLIOptions.taskName() === "build") {
```

Comment:
I added the comment there to emphasize that the build task can be executed by the run task, but I'll happily remove it. For me the order of the conditions doesn't really affect the comment and this order mirrors the one in run.ts, but I'll swap them if you want.